kfs-purap/src/main/java/edu/arizona/kfs/module/purap/service/impl/TaxReporting1099ServiceImpl.java

```diff
@@ -0,0 +1,1806 @@
+		List<Payee> payees = (List<Payee>) businessObjectService.findMatching(Payee.class, criteria);
+
+		if ((payees != null) && !payees.isEmpty()) {
+			retval = payees.get(0);
+		}
+
+		return retval;
+	}
+
+	private Payee getPayeeByVendorNum(VendorDetail vendor, Integer taxYear) {
+		Payee retval = getPayeeByVendorNum(vendor.getVendorNumber(), taxYear);
+
+		// if we did not find payee then create a new one
+		if (retval == null) {
+			retval = createNewPayee(taxYear, vendor);
+		}
+
+		return retval;
+	}
+
+	private List<VendorDetail> getVendors(String ownershipCode, boolean isAllow) {
+		List<VendorDetail> retval = null;
+		String vendorOwnershipCode = null;
+		String vendorOwnershipCategoryCode = null;
+
+		if (!ownershipCode.contains("=")) {
+			vendorOwnershipCode = ownershipCode;
+		} 
+		else {
+			int pos = ownershipCode.indexOf('=');
+			vendorOwnershipCode = ownershipCode.substring(0, pos);
+			vendorOwnershipCategoryCode = ownershipCode.substring(pos + 1);
+		}
+
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("vendor ownership code: " + vendorOwnershipCode);
+			LOG.debug("vendor ownership category code: " + vendorOwnershipCategoryCode);
+		}
+
+		retval = taxReporting1099Dao.getVendors(vendorOwnershipCode, vendorOwnershipCategoryCode, isAllow);
+
+		if (LOG.isInfoEnabled()) {
+			int cnt = 0;
+			if ((retval != null) && !retval.isEmpty()) {
+				cnt = retval.size();
+			}
+
+			LOG.info("found " + cnt + " vendors with ownership code=" + vendorOwnershipCode);
+		}
+
+		return retval;
+	}
+
+	private PdfReader getPdfReader(Integer year) {
+		PdfReader retval = null;
+		FileInputStream fispdf = null;
+		try {
+			fispdf = new FileInputStream(get1099PdfFile(year));
+			retval = new PdfReader(fispdf);
+		} 
+		catch (Exception ex) {
+			throw new RuntimeException(ex);
+		}
+
+		finally {
+			try {
+				fispdf.close();
+			} 
+			catch (Exception ex) {
+			}
+
+		}
+
+		return retval;
+	}
+
+	private File get1099PdfFile(Integer year) {
+		File retval = null;
+
+		String fname = pdfDirectory + "/" + TaxPropertyConstants.PDF_1099_FILENAME_PREFIX + "-" + year + ".pdf";
+
+		LOG.info("pdfDirectory: " + pdfDirectory);
+
+		retval = new File(fname);
+
+		if (!retval.exists() || !retval.isFile() || retval.length() <= 0) {
+			throw new RuntimeException("The IRS 1099 Misc PDF [" + retval.getAbsolutePath() + "] is not available in the file system.");
+		}
+
+		LOG.info("1099 PDF filename = " + retval.getAbsolutePath());
+
+		return retval;
+	}
+
+	@Override
+	public List<Payee> loadPayees(Integer year) {
+		Map<String, Object> criteria = new HashMap<String, Object>();
+		criteria.put(TaxPropertyConstants.TAX_YEAR, year);
+		criteria.put(TaxPropertyConstants.EXCLUDE_INDICATOR, Boolean.FALSE);
+		return loadPayee1099Information(year, (List<Payee>) businessObjectService.findMatching(Payee.class, criteria));
+	}
+
+	private Payee loadPayee(Integer year, Payee payee) {
+		Payee retval = null;
+
+		// lets get all the payees for this a
```

Comment:
This is what the comment was in the original code. This has been revised based on what the code looks like it's doing.